internal/handlers/sqlite/msg_delete.go

```diff
@@ -16,12 +16,129 @@ package sqlite
 
 import (
 	"context"
+	"errors"
 
+	"github.com/FerretDB/FerretDB/internal/backends"
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgDelete implements HandlerInterface.
 func (h *Handler) MsgDelete(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	return nil, notImplemented(must.NotFail(msg.Document()).Command())
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	params, err := common.GetDeleteParams(document, h.L)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	var deleted int32
+	var delErrors commonerrors.WriteErrors
+
+	db := h.b.Database(params.DB)
+	defer db.Close()
+
+	coll := db.Collection(params.Collection)
+
+	// process every delete filter
+	for i, deleteParams := range params.Deletes {
+		del, err := execDelete(ctx, coll, deleteParams.Filter, deleteParams.Limited)
+		if err == nil {
+			deleted += del
+			continue
+		}
+
+		delErrors.Append(err, int32(i))
```

Comment:
If we had four delete params, and let's say the first and third had errors, for this append, we will pass `i` that are `0` and `2`, is it correct?

Reply:
Yes, seems so. I haven't changed this logic compared to the Postgres handler.